doc: vsc: update modifying application page (#7523)
nrfconnect/sdk-nrf: doc/nrf/gs_modifying.rst

To quickly test different configuration options, or to build your application in different variants, you can update the :file:`.config` file in the build directory.
Changes are picked up immediately, and you do not need to re-open the project in SES.

> While it is possible to edit the :file:`.config` file directly, you should use the nRF Kconfig extension in |VSC| or a tool like menuconfig or guiconfig to update it.

Reviewer: ```suggestion
While it is possible to edit the :file:`.config` file directly, you should use the nRF Kconfig GUI in |VSC| or a tool like menuconfig or guiconfig to update it.
```

These tools present all available options and allow you to select the ones that you need.

The nRF Kconfig extension in |VSC| organizes the Kconfig options in a hierarchical list and lets you select the desired options.

> To save the changes made using the nRF Kconfig extension, click the :guilabel:`Save` button.

Reviewer: ```suggestion
To save the changes made using the nRF Kconfig GUI, click the :guilabel:`Save` button.
```

Read the `nRF Kconfig`_ documentation for more information.

To edit the file in SES, select :guilabel:`Project` > :guilabel:`Configure nRF Connect SDK Project`.
If your application contains more than one image (see :ref:`ug_multi_image`), you must select the correct target.
To configure the parent image (the main application), select :guilabel:`menuconfig`.
The other options allow you to configure the child images.

See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how to run menuconfig or guiconfig.

> To locate a specific configuration option, use the filter (:guilabel:`Search modules` field in nRF Kconfig or :guilabel:`Jump to` in menuconfig and guiconfig).

Reviewer: ```suggestion
To locate a specific configuration option, use the filter (:guilabel:`Search modules` field in the nRF Kconfig GUI or :guilabel:`Jump to` in menuconfig and guiconfig).
```

While it is possible to edit the :file:`.config` file directly, you should use SES or a tool like menuconfig or guiconfig to update it.
These tools present all available options and allow you to select the ones that you need.

To edit the file in |VSC|, use the nRF Kconfig extension.

> This extension organizes the Kconfig option in a hierarchical list and lets you select and deselect the desired options.

Reviewer: ```suggestion
This extension organizes the Kconfig options in a hierarchical list and lets you select the desired options.
```

> To save the changes made using the nRF Kconfig extension, click the :guilabel:`Save` button.

Reviewer: We have three different save options actually. Are we assuming here that the user is selecting their kconfig options for the first time and needs to just save them to a new .conf file?

Author: I'm using the differentiation mentioned in the VSC docs. As we're talking about temporary changes, I'm mentioning "Save". In line 264, where we mention permanent changes, I mention "Save to file".

> Read the `nRF Kconfig`_ documentation for more information.

Reviewer 2: This whole paragraph should be merged with the previous one about menuconfig, I think. This is describing "menuconfig", but for VS Code. The nRF Kconfig extension does a lot more than this Kconfig menu interface, so I think we should be referring to this feature as the VS Code Kconfig menu (or whatever we ended up calling it, @Avalei )

Reviewer: In a conversation on May 9th we decided that we will use "_nRF Kconfig when referring to the extension as a whole, and "the nRF Kconfig GUI" when talking about the interface itself._"

To edit the file in SES, select :guilabel:`Project` > :guilabel:`Configure nRF Connect SDK Project`.
If your application contains more than one image (see :ref:`ug_multi_image`), you must select the correct target.
To configure the parent image (the main application), select :guilabel:`menuconfig`.
The other options allow you to configure the child images.

See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how to run menuconfig or guiconfig.

> To locate a specific configuration option, use the filter (:guilabel:`Jump To Item` in nRF Kconfig or :guilabel:`Jump to` in menuconfig and guiconfig).

Reviewer: The "Jump to Item" options is only visible on options you have already enabled and which are showing up in the Changes tab. If you want to search for a new kconfig option to enable, it's better to type in the name of the option in the search field.

Author: Edited. Please check.

The file will be reloaded when CMake re-runs.
This will happen automatically when the application is rebuilt, but can also be requested manually by using the :guilabel:`Project` > :guilabel:`Run CMake...` option.

If you work with |VSC|, you can use one of the following options:

* Select an extra Kconfig fragment file when `Building an application`_.

> * Edit the Kconfig options using the `nRF Kconfig`_ extension and save changes permanently to an existing or new :file:`prj.conf` file.

Author: Edits applied. Please re-check.

The file will be reloaded when CMake re-runs.
This will happen automatically when the application is rebuilt, but can also be requested manually by using the :guilabel:`Project` > :guilabel:`Run CMake...` option.

If you work with |VSC|, you can use one of the following options:

* Select an extra Kconfig fragment file when `Building an application`_.
* Edit the Kconfig options using the `nRF Kconfig`_ extension.

>   To save changes permanently to the :file:`prj.conf` file, click the :guilabel:`Save to file` button.

Reviewer: Is "permanently" the right word to use here? I believe both "Save to file" and "Save (minimal)" are both technically permanent saves it just depends on if there is an existing .conf file already or not. @trond-snekvik is that right? We might want to provide a little more context around the saving options for this section.

- Save: saves temporary changes to the <build>/zephyr/.conf file.
- Save to file: saves changes to an existing .conf file.
- Save (minimal): saves the smallest subset of settings to a new .conf file.

Reviewer 2: I would say "permanently" is the right concept, but I don't know if it's the right phrase. It's definitely "non-temporary" at least 😄 Agree with the rest!

Author: Hm, on second thoughts, maybe I don't need to describe the UI button here, so I rephrased for simplicity. But I did keep the "Save" in line 228.

  Click :guilabel:`File` > :guilabel:`Open nRF Connect SDK project`, select :guilabel:`Extended Settings`, and specify the options in the :guilabel:`Extra CMake Build Options` field.
  This field is prepopulated with the global CMake options, and you can modify them, remove them, or add to them for the current project.

If you work with the |VSC| extension, you can specify project-specific CMake options when you add the build configuration for a new |NCS| project.

> See `Building an application`_ in the nRF Connect for Visual Studio Code documentation.

Reviewer: ```suggestion
See `Building an application`_ in the |VSC| documentation.
```
